Userland/Applications/Presenter/SlideObject.cpp

```diff
@@ -0,0 +1,158 @@
+/*
+ * Copyright (c) 2022, kleines Filmröllchen <filmroellchen@serenityos.org>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include "SlideObject.h"
+#include "AK/kmalloc.h"
+#include "LibCore/Stream.h"
+#include "LibGUI/Margins.h"
+#include "LibGfx/Forward.h"
+#include "LibGfx/Orientation.h"
+#include "LibImageDecoderClient/Client.h"
+#include <AK/JsonObject.h>
+#include <AK/RefPtr.h>
+#include <LibCore/Object.h>
+#include <LibGfx/Font/FontDatabase.h>
+#include <LibGfx/Painter.h>
+#include <LibGfx/Size.h>
+#include <LibGfx/TextWrapping.h>
+#include <utility>
+
+ErrorOr<NonnullRefPtr<SlideObject>> SlideObject::parse_slide_object(JsonObject const& slide_object_json, NonnullRefPtr<GUI::Window> window)
+{
+    auto image_decoder_client = TRY(ImageDecoderClient::Client::try_create());
+
+    auto const& maybe_type = slide_object_json.get("type"sv);
+    if (!maybe_type.is_string())
+        return Error::from_string_view("Slide object must have a type"sv);
+
+    auto type = maybe_type.as_string();
+    RefPtr<SlideObject> object;
+    if (type == "text"sv) {
+        object = TRY(try_make_ref_counted<Text>());
+    } else if (type == "image"sv) {
+        object = TRY(try_make_ref_counted<Image>(image_decoder_client, window));
+    } else {
+        return Error::from_string_view("Unsupported slide object type"sv);
+    }
+
+    slide_object_json.for_each_member([&](auto const& key, auto const& value) {
+        // Ignore failures:
+        // Pro: No need to check legal keys in advance, no need to check types, ...
+        // Con: Illegal properties are silently dropped
+        object->set_property(key, value);
+    });
+
+    return object.release_nonnull();
+}
+
+SlideObject::SlideObject()
+{
+    REGISTER_RECT_PROPERTY("rect", rect, set_rect);
+}
+
+void SlideObject::paint([[maybe_unused]] Gfx::Painter& painter, [[maybe_unused]] Gfx::FloatSize display_scale) const { }
```

Comment:
You can just not mention the name to show if you're not using it
`void SlideObject::paint(Gfx::Painter&, Gfx::FloatSize) const { }`